### PR TITLE
Spark3: Support for `EXPLAIN` statement

### PR DESCRIPTION
**Pull request by R7L208**
### Brief summary of the change made

Replace `ExplainStatementSegment` from ANSI to allow for additional parameters and an expanded list of explainable statements.

### Are there any other side effects of this change that we should be aware of?
Removed `parse_grammar` from `ExplainStatementSegment` in ANSI and re-assigned `parse_grammar` to `match_grammar` in Postgres and Snowflake dialects.

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_spark3.py [hunk 1]
```diff
@@ -1679,6 +1679,61 @@ class TransformClauseSegment(BaseSegment):
...
+            Ref("AlterViewStatementSegment"),
+            Ref("CreateDatabaseStatementSegment"),
+            Ref("CreateFunctionStatementSegment"),
+            Ref("CreateTableStatementSegment"),
+            Ref("CreateViewStatementSegment"),
+            ansi_dialect.get_segment("DropDatabaseStatementSegment"),
+            Ref("DropFunctionStatementSegment"),
+            ansi_dialect.get_segment(
+                "DropViewStatementSegment",
+            ),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_spark3.py:1710]: Why are these not just Refs? like the rest?

```suggestion
            Ref("DropDatabaseStatementSegment"),
            Ref("DropFunctionStatementSegment"),
            Ref("DropViewStatementSegment"),
```

File: src/sqlfluff/dialects/dialect_spark3.py [hunk 2]
```diff
@@ -1679,6 +1679,61 @@ class TransformClauseSegment(BaseSegment):
...
+            Ref("LoadDataSegment"),
+            Ref("InsertOverwriteDirectorySegment"),
+            Ref("InsertOverwriteDirectoryHiveFmtSegment"),
+        ],
+    )
+
+    match_grammar = StartsWith("EXPLAIN")
+
+    parse_grammar = Sequence(
+        "EXPLAIN",
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_spark3.py:1725]: This does nothing as they both start with `EXPLAIN`. See it's like this in ANSI but still think we should clean up (would take the ANSI clean up at same time if you like!)

```suggestion
    match_grammar = Sequence(
        "EXPLAIN",
```

File: src/sqlfluff/dialects/dialect_spark3.py [hunk 3]
```diff
@@ -1679,6 +1679,61 @@ class TransformClauseSegment(BaseSegment):
...
+    https://spark.apache.org/docs/latest/sql-ref-syntax-qry-explain.html
+    """
+
+    type = "explain_statement"
+
+    explainable_stmt = ansi_dialect.get_segment(
+        "ExplainStatementSegment"
+    ).explainable_stmt.copy(
+        insert=[
+            Ref("AlterDatabaseStatementSegment"),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_spark3.py:1699]: Any reason this shouldn't just be all statements (i.e. `Ref("StatementSegment"),`)?

**Review by tunetheweb — CHANGES_REQUESTED** [review]
A few comments

File: src/sqlfluff/dialects/dialect_spark3.py (hunk 3 above)
Comment (R7L208) [reply src/sqlfluff/dialects/dialect_spark3.py:1699]: Perfect! I had looked for something like that and missed it.

> Why are these not just Refs? like the rest?
From the comment above, Using `StatementSegment` would resolve that as well.

File: src/sqlfluff/dialects/dialect_spark3.py (hunk 2 above)
Comment (R7L208) [reply src/sqlfluff/dialects/dialect_spark3.py:1725]: I'll clean this up in Spark and ANSI

**Comment by R7L208** [comment]
Updated PR description to reflect most recent change of removing `parse_grammar` from `ExplainStatementSegment`

**Review by tunetheweb — APPROVED** [review]
Thanks for adding those changes!

**Pull request merged**